presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -191,6 +188,194 @@ This table can then be queried in Presto::
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Access ORC columns using names from the file. By default, columns in ORC files are accessed by their ordinal position in the Hive table definition. Setting this property allows to use columns names recorded in the ORC file instead.
+
+
+.. _parquet-optimized-reader:
+
+``hive.parquet-optimized-reader.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** *Deprecated* Serves as default value for ``parquet_optimized_reader_enabled`` session property. Enables number of reader improvements introduced by alternative parquet implementation. The new reader supports vectorized reads, lazy loading, and predicate push down, all of which make the reader more efficient and typically reduces wall clock time for a query. However as the code has changed significantly it may or may not introduce some minor issues, so it can be disabled if some  problems with environment are noticed. This property enables/disables all optimizations except of predicate pushdown as it is managed by ``hive.parquet-predicate-pushdown.enabled`` property.
+
+
+``hive.parquet-predicate-pushdown.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** *Deprecated* Serves as default value for ``parquet_predicate_pushdown_enabled`` sesssion property. See :ref:`hive.parquet-optimized-reader.enabled<parquet-optimized-reader>`.
+
+
+``hive.parquet.use-column-names``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Access Parquet columns using names from the file. By default, columns in Parquet files are accessed by their ordinal position in the Hive table definition. Setting this property allows to use columns names recorded in the Parquet file instead.
+
+
+``hive.s3.max-connections``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``500``
+ * **Description:** This value the maximum number of connections to S3. How many connection to S3 cluster may be open at the same time by the S3 driver. Higher value may increase network utilization when cluster is used on high speed network. However higher value relies more on S3 servers being well configured for high parallelism.
+
+
+``hive.s3.multipart.min-file-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size, at least ``16 MB``)
+ * **Default value:** ``16 MB``
+ * **Description:** Minimum file size for an S3 multipart upload. This property describes how big file must be to be uploaded to S3 cluster using multipart feature. Amazon recommendation is to use ``100 MB`` value here, however lower value may allow to increase upload parallelism and can decrease ``data lost``/``data sent`` ratio in unstable network conditions.
```

Comment:
This property describes how big a file must be to be uploaded to an S3 cluster using the multipart upload feature.  Amazon recommends using ``100 MB``, but a lower value may increase upload parallelism and decrease the ``data lost``/``data sent`` ratio in unstable network conditions.